tests/unit/room.spec.js

```diff
@@ -0,0 +1,155 @@
+import Vue from 'vue';
+import { shallowMount, createLocalVue } from '@vue/test-utils';
+import Vuex from 'vuex';
+import VueEvents from 'vue-events';
+import Vuetify from 'vuetify';
+import VueSlider from 'vue-slider-component';
+import Room from '@/views/Room';
+
+// HACK: import globally to prevent it from yelling at us
+// https://github.com/vuetifyjs/vuetify/issues/4964
+Vue.use(Vuetify);
+
+const localVue = createLocalVue();
+localVue.use(Vuex);
+localVue.use(VueEvents);
+localVue.component('VueSlider', VueSlider);
+
+const $route = {
+  path: 'http://localhost:8080/room/example',
+  params: {
+    roomId: 'example',
+  },
+};
+
+describe('Room UI spec', () => {
+  let wrapper;
+  let store;
+
+  beforeEach(() => {
+    store = new Vuex.Store({
+      state: {
+        socket: {
+          isConnected: false,
+          message: '',
+          reconnectError: false,
+        },
+        joinFailureReason: null,
+        production: true,
+        room: {
+          name: "example",
+          title: "",
+          description: "",
+          isTemporary: false,
+          currentSource: {},
+          queue: [],
+          isPlaying: false,
+          playbackPosition: 0,
+        },
+      },
+    });
+
+    wrapper = shallowMount(Room, {
+      store,
+      localVue,
+      mocks: {
+        $route,
+        $connect: jest.fn(),
+        $disconnect: jest.fn(),
+      },
+      stubs: ['youtube'],
+    });
+  });
+
+  it('should render room title in permanent rooms, if the room has one', () => {
+    store.state.room.title = 'Example Room';
+    store.state.room.isTemporary = false;
+    const roomTitle = wrapper.find('h1');
```

Comment:
This is not `h1` in my PR for editable room names. Maybe consider #83 a blocker.